domain/src/main/java/com/teamagam/gimelgimel/domain/layers/OnVectorLayerListingClickInteractor.java

```diff
@@ -0,0 +1,43 @@
+package com.teamagam.gimelgimel.domain.layers;
+
+import com.google.auto.factory.AutoFactory;
+import com.google.auto.factory.Provided;
+import com.teamagam.gimelgimel.domain.base.executor.ThreadExecutor;
+import com.teamagam.gimelgimel.domain.base.interactors.BaseDataInteractor;
+import com.teamagam.gimelgimel.domain.base.interactors.DataSubscriptionRequest;
+import com.teamagam.gimelgimel.domain.layers.entitiy.VectorLayerPresentation;
+import com.teamagam.gimelgimel.domain.layers.entitiy.VectorLayerVisibilityChange;
+import com.teamagam.gimelgimel.domain.layers.repository.VectorLayersVisibilityRepository;
+import java.util.Collections;
+import rx.Observable;
+
+@AutoFactory
+public class OnVectorLayerListingClickInteractor extends BaseDataInteractor {
+
+  private final VectorLayersVisibilityRepository mVectorLayersVisibilityRepository;
+  private final VectorLayerPresentation mVectorLayerPresentation;
+
+  public OnVectorLayerListingClickInteractor(@Provided ThreadExecutor threadExecutor,
+      @Provided VectorLayersVisibilityRepository vectorLayersVisibilityRepository,
+      VectorLayerPresentation vectorLayerPresentation) {
+    super(threadExecutor);
+    mVectorLayersVisibilityRepository = vectorLayersVisibilityRepository;
+    mVectorLayerPresentation = vectorLayerPresentation;
+  }
+
+  @Override
+  protected Iterable<SubscriptionRequest> buildSubscriptionRequests(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+    SubscriptionRequest setVisibilityRequest =
+        factory.create(Observable.just(mVectorLayerPresentation),
```

Comment:
When rebasing on Rx2, I think maybe you should use the new SIGNAL convention. But only if it doesn't reduce readability.